include/boost/core/bit.hpp

```diff
@@ -47,12 +47,19 @@
 # if __has_builtin(__builtin_bit_cast)
 #  define BOOST_CORE_HAS_BUILTIN_BIT_CAST
 # endif
+# if __has_builtin(__builtin_bswap16)
+#  define BOOST_CORE_HAS_BUILTIN_BSWAP16
+# endif
 #endif
 
-#if defined(BOOST_MSVC) && BOOST_MSVC >= 1926
+#if !defined(BOOST_CORE_HAS_BUILTIN_BIT_CAST) && (defined(BOOST_MSVC) && BOOST_MSVC >= 1926)
 #  define BOOST_CORE_HAS_BUILTIN_BIT_CAST
 #endif
 
+#if !defined(BOOST_CORE_HAS_BUILTIN_BSWAP16) && (defined(BOOST_GCC_VERSION) && BOOST_GCC_VERSION >= 40800)
```

Comment:
Why BOOST_GCC_VERSION instead of BOOST_GCC?

Reply:
Well? This looks OK to merge except for this bit.

As far as I can see, `BOOST_GCC_VERSION` is mostly equivalent to `BOOST_GCC`, but is undocumented.

Seems to me that if we want "real GCC 4.8 or above", this should be `BOOST_GCC`, and if we want "GCC-4.8-compatible", this should be `__GNUC__ * 100 + __GNUC_MINOR__ >= 408`.